app/scienceapi/events/admin.py

```diff
@@ -1,5 +1,46 @@
 from django.contrib import admin
+from django.utils import timezone
+from django.utils.http import unquote
 
+import pytz
 from .models import Event
+from .forms import EventAdminForm
 
-admin.site.register(Event)
+@admin.register(Event)
+class EventAdmin(admin.ModelAdmin):
+
+    form = EventAdminForm
+
+    def save_model(self, request, obj, form, change):
+        print('hello')
+        tz = pytz.timezone(obj.timezone)
+
+        # First converting to naive timezone, otherwise localize wont work.
+        start = obj.starts_at.replace(tzinfo=None)
+        end = obj.ends_at.replace(tzinfo=None)
+
+        # Using the user selected timezone
+        obj.starts_at = tz.localize(start)
+        obj.ends_at = tz.localize(end)
+        super().save_model(request, obj, form, change)
+
+    def get_object(self, request, object_id, from_field):
+        obj = super().get_object(request, object_id, from_field)
+        
```

Comment:
@gideonthomas so for changing timezone before rendering I overrode `get_object` this method is not listed in the api but in source code, so i don't know how correct this is.